CMakeLists.txt

```diff
@@ -226,6 +226,12 @@ if(WITH_LOGS_PREVIEW)
   add_definitions(-DENABLE_LOGS_PREVIEW)
 endif()
 
+option(WITH_ASYNC_EXPORT_PREVIEW "Whether enable async export" OFF)
+
+if(WITH_ASYNC_EXPORT_PREVIEW)
```

Comment:
We should also add INTERFACE definitions into [api/CMakeLists.txt](api/CMakeLists.txt).

```cmake
if(WITH_LOGS_PREVIEW)
  target_compile_definitions(opentelemetry_api INTERFACE ENABLE_ASYNC_EXPORT)
endif()
```

So that when we use optentelemetry-cpp as a cmake module(`find_package(opentelemetry-cpp CONFIG)`) , this definition can be auto added into all targets that direct or indirectly depend on `opentelemetry-cpp`.

I find there are severval missing definitions in [api/CMakeLists.txt](api/CMakeLists.txt) and be add by `add_definitions()`. All definitions should be added by `target_compile_definitions(opentelemetry_api INTERFACE ...)` to be expoted by cmake.
We should avoid to use `add_definitions(...)` unless the definitions is only used by unit tests, should I create another issue to fix it? @lalitb @ThomsonTan

Reply:
Added.